crates/gitbutler-branch-actions/src/remote.rs

```diff
@@ -60,32 +62,50 @@ pub struct RemoteCommit {
     pub conflicted: bool,
 }
 
-/// Return information on all local branches, while skipping gitbutler-specific branches in `refs/heads`.
-///
-/// Note to be confused with `list_branches()`, which is used for the new branch listing.
+/// Finds all branches matching a given name, which can be at most one local branch,
+/// and any number of branches (on different remotes).
 ///
 /// # Previous notes
 /// For legacy purposes, this is still named "remote" branches, but it's actually
 /// a list of all the normal (non-gitbutler) git branches.
-pub fn list_local_branches(ctx: &CommandContext) -> Result<Vec<RemoteBranch>> {
+pub fn find_git_branches(ctx: &CommandContext, branch_name: &str) -> Result<Vec<RemoteBranch>> {
     let default_target = default_target(&ctx.project().gb_dir())?;
-
-    let mut remote_branches = vec![];
-    let remotes = ctx.repository().remotes()?;
-    for (branch, _) in ctx
-        .repository()
-        .branches(None)
-        .context("failed to list remote branches")?
-        .flatten()
+    let repo = ctx.repository();
+    let mut branches = vec![];
+    let remotes_raw = repo.remotes()?;
+    let remotes = remotes_raw.iter().flatten().collect_vec();
```

Comment:
Why do we need to flatten the remotes array? Isn't the return type `Result<Vec<String>>`?

Reply:
We need to flatten it since remotes_raw is a git2::StringArray, which converts into `Vec<Option<&str>>`.

Reply:
I see